core/sql/parser/sqlparser.y

```diff
@@ -24909,7 +24946,6 @@ create_table_as_token: TOK_AS
 
 /* type pElemDDL */
 table_definition_body : table_element_list
-                      | like_definition
                       | external_table_definition
 
```

Comment:
This is intentional. The problem is that including like_definition in table_definition_body allows things like STORE BY and several other clauses to follow the LIKE option. But the LIKE code just ignores them all. That is, the original implementation was incomplete. I wanted to make the unsupported syntax invalid.